packages/@aws-cdk/core/lib/runtime.ts

```diff
@@ -205,7 +313,7 @@ export type Validator = (x: any) => ValidationResult;
  */
 export function canInspect(x: any) {
   // Note: using weak equality on purpose, we also want to catch undefined
-  return (x != null && !isCloudFormationIntrinsic(x));
+  return (x != null && !isCloudFormationIntrinsic(x) && !isResolvableObject(x));
```

Comment:
Mmm. Did you run into an issue with this? iirc `canInspect` should be called in a part in the rendering cycle where it shouldn't be able to run into Tokens... (but I might be wrong)

Reply:
Yes, this was my mistake. I was calling the validators inside the generated `<struct>FromCloudFormation` methods, which is not needed, as they will be called anyway during template synthesis. I've reverted this change (and I no longer call the validators in the generated `<struct>FromCloudFormation` methods).